### ISSUE TITLE
Firefox Fetch implementation is not working, JSON.parse errors

### ISSUE THREAD

**Issue by feedthebayer** [pixelhandler/ember-jsonapi-resources#43]
To reproduce, simply clone and setup this test repo and try to load in Firefox: https://github.com/pixelhandler/jr-test

With logging enabled you get this in the console:
```console
Attempting URL transition to / vendor.js:59372:7
generated -> route:application Object { fullName: "route:application" } vendor.js:36898:1
Preparing to transition from '' to ' index' vendor.js:36658:9
Transition #0: application: calling beforeModel hook vendor.js:59369:7
Transition #0: application: calling deserialize hook vendor.js:59369:7
Transition #0: application: calling afterModel hook vendor.js:59369:7
Transition #0: index: calling beforeModel hook vendor.js:59369:7
Transition #0: index: calling deserialize hook vendor.js:59369:7
Error while processing route: index JSON.parse: unexpected character at line 1 column 1 of the JSON data vendor.js:37221:5
Transition #0: index: transition was aborted
```

@pixelhandler said:
> I tried to debug but seems like the issue is somewhere in the polyfill the workaround is to use XHR instead of fetch which requires an alternative implementation of the adapter.fetch method which will take some time for me to get too.

**Comment by pixelhandler** [comment]
@feedthebayer I'll work on a mixin that only uses XHR instead of window.fetch for the application adapter. 

The current issue in firefox is that window.fetch ends up with the html page instead of the JSON from the api server. the jr-test project for ember-jsonapi-resource reproduces the bug.

In the adapter.fetch method the `catch` throws an error, inspecting the request shows that the html page is the response instead of the expected JSON.

**Comment by pixelhandler** [comment]
@feedthebayer here is a workaround you can try: https://github.com/pixelhandler/ember-jsonapi-resources/compare/use-xhr-instead-of-fetch adapter uses a flag for `useAjax` and calls adapter#ajax instead of using window.fetch (w/ polyfill)

**Comment by pixelhandler** [comment]
@feedthebayer try out 0.6.3

**Comment by feedthebayer** [comment]
@pixelhandler It works in Firefox right out of the box! Thank you!!

**Comment by feedthebayer** [comment]
@pixelhandler I'm running into this Firefox issue again. I can't `POST` or `PATCH` from Firefox (works fine in Chrome). `DELETE` does work though.

I get a `400 Bad Request` error from the API with this error detail: `The required parameter, data, is missing.`

**Comment by pixelhandler** [comment]
@feedthebayer ok I'll look into that soon

**Comment by pixelhandler** [comment]
@feedthebayer I published 0.7.1 with bugfix for xhr data.

**Comment by feedthebayer** [comment]
It works! Thanks!!
